test/NuGet.Core.Tests/NuGet.Packaging.Test/DependenciesGroupsForEachTFMRuleTests.cs

```diff
@@ -0,0 +1,446 @@
+
+            // Assert
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5128));
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5130));
+        }
+
+        [Fact]
+        public void GenerateWarnings_PackageWithDependenciesInRefHasExactMatch_ShouldNotWarn()
+        {
+            var frameworks = new NuGetFramework[]
+            {
+                FrameworkConstants.CommonFrameworks.Net2,
+                FrameworkConstants.CommonFrameworks.Net35,
+                FrameworkConstants.CommonFrameworks.Net4,
+                FrameworkConstants.CommonFrameworks.Net45,
+                FrameworkConstants.CommonFrameworks.NetStandard10,
+                FrameworkConstants.CommonFrameworks.NetStandard13,
+                FrameworkConstants.CommonFrameworks.NetStandard20,
+            };
+
+            //Arrange
+            var fileStrings = new string[]
+            {
+                "ref/net20/test.dll",
+                "ref/net35/test.dll",
+                "ref/net40/test.dll",
+                "ref/net45/test.dll",
+                "ref/netstandard1.0/test.dll",
+                "ref/netstandard1.3/test.dll",
+                "ref/netstandard2.0/test.dll",
+            };
+
+            // Act
+            var rule = new DependenciesGroupsForEachTFMRule();
+            var (compat, file, nuspec) = rule.Categorize(fileStrings, frameworks);
+            var issues = rule.GenerateWarnings(compat, file, nuspec).ToList();
+
+            // Assert
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5128));
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5130));
+        }
+
+        [Fact]
+        public void GenerateWarnings_PackageWithoutDependenciesForEachTFMInLib_ShouldThrowOneWarningCode()
+        {
+            //Arrange
+            var frameworks = new NuGetFramework[]
+            {
+                FrameworkConstants.CommonFrameworks.Net2,
+                FrameworkConstants.CommonFrameworks.Net35,
+                FrameworkConstants.CommonFrameworks.Net4,
+                FrameworkConstants.CommonFrameworks.Net45,
+                FrameworkConstants.CommonFrameworks.NetStandard10,
+                FrameworkConstants.CommonFrameworks.NetStandard13,
+                FrameworkConstants.CommonFrameworks.NetStandard20,
+            };
+
+            var fileStrings = new string[]
+            {
+                "lib/net20/test.dll",
+                "lib/net35/test.dll",
+                "lib/net40/test.dll"
+            };
+
+            // Act
+            var rule = new DependenciesGroupsForEachTFMRule();
+            var (compat, file, nuspec) = rule.Categorize(fileStrings, frameworks);
+            var issues = rule.GenerateWarnings(compat, file, nuspec).ToList();
+
+            // Assert
+            var testStringExact = AnalysisResources.DependenciesGroupsForEachTFMHasNoExactMatch + "\n" +
```

Comment:
One line having a `\n` at the end, but the other using `\r\n` is a code smell. My previous comment about using the string build for all string, and not using `String.Join` will fix it. But in the future when you see something like this, you should investigate because something's not right.